font-codegen/src/parsing.rs

```diff
@@ -636,25 +682,34 @@ impl FieldType {
             }
         }
 
-        if last.ident != "BigEndian" {
-            return Ok(FieldType::Other { typ: path.clone() });
+        if is_wellknown_scalar(last) {
+            return Ok(FieldType::Scalar {
+                typ: last.ident.clone(),
+            });
+        }
+
+        if is_wellknown_struct(last) {
+            return Ok(FieldType::Struct {
+                typ: last.ident.clone(),
+            });
         }
 
-        let inner = get_single_generic_type_arg(&last.arguments)?;
-        let last = inner.segments.last().unwrap();
         if ["Offset16", "Offset24", "Offset32"].contains(&last.ident.to_string().as_str()) {
             let target = get_offset_target(&last.arguments)?;
-            Ok(FieldType::Offset {
+            return Ok(FieldType::Offset {
                 typ: last.ident.clone(),
                 target,
-            })
-        } else if last.arguments.is_empty() {
-            Ok(FieldType::Scalar {
-                typ: last.ident.clone(),
-            })
-        } else {
-            Err(syn::Error::new(last.span(), "unexpected arguments"))
+            });
         }
+
+        // We'll figure it out later, what could go wrong?
+        if !last.arguments.is_empty() {
+            panic!("Whatever shall we do with {}", quote! { #last })
```

Comment:
I personally would prefer to keep returning an error here? And if we're going to decide that we prefer panicking to returning an error, we should probably do that consistently, and then we can change all the type signatures to no longer return `Result`, and then I can also rip out all of the machinery involved in printing spans in the input files.

So maybe: if we decide we want to panic instead of printing error locations (and it isn't clear to me at this point that this is a clear improvement?) we could do a PR that goes through and does that consistently?

Reply:
+1 to keeping the panic vs error out of this, I will change them to return error.